### PR TITLE
Fix infinite loop case in autocomplete.js

### LINKED ISSUE #1355: If you enter "{{" into the labor category search, your browser/computer will run out of memory
Seems like it gets stuck in some kind of a loop. This was found during the penetration testing.

Pretty nasty!

### PR DESCRIPTION
**Pull request by jseppi**
Fixes #1355

In the case of `searchStr` comprising only special characters (like `"{{"`) the `while` loop in `appendHighlightedTerm` would never exit because the `sanitizedSearchString` was the empty string, so the regular expression built from it would always return a match.

The code in `autocomplete.js` is still a bit dubious, and I don't love that it is using jQuery. I bet we could could either find or write a better React autocomplete.

### CODE REVIEW

File: frontend/source/js/data-explorer/autocomplete.js [hunk 1]
```diff
@@ -66,6 +71,7 @@ export function initialize(el, {
       }, (error, result) => {
         autoCompReq = null;
         if (error) { return done([]); }
+        debugger;
```
Comment (toolness) [inline frontend/source/js/data-explorer/autocomplete.js]: I guess this should be removed?

**Comment by jseppi** [comment]
I can bust up a test right quick

**Comment by jseppi** [comment]
boom! also added some tests for the basic working case.

**Comment by jseppi** [comment]
ready for review again, @toolness

**Pull request merged**